src/main/java/com/dre/brewery/configuration/ConfigManager.java

```diff
@@ -0,0 +1,254 @@
+     * @param <T> The type of the config
+     */
+    public static <T extends AbstractOkaeriConfigFile> T getConfig(Class<T> configClass) {
+        try {
+            for (var mapEntry : LOADED_CONFIGS.entrySet()) {
+                if (mapEntry.getKey().equals(configClass)) {
+                    return (T) mapEntry.getValue();
+                }
+            }
+            return createConfig(configClass);
+        } catch (Throwable e) {
+            BreweryPlugin.getInstance().errorLog("Something went wrong trying to load a config file... " + configClass.getSimpleName(), e);
+            BreweryPlugin.getInstance().warningLog("Resolve the issue in the file and run /brew reload");
+            return null;
+        }
+    }
+
+    /**
+     * Replaces a config instance in the LOADED_CONFIGS map with a new instance of the same class
+     * @param configClass The class of the config to replace
+     * @param <T> The type of the config
+     */
+    public static <T extends AbstractOkaeriConfigFile> void newInstance(T configClass) {
+        LOADED_CONFIGS.put(configClass.getClass(),
+                createConfig(configClass.getClass(),
+                        configClass.getBindFile().getFileName().toString(),
+                        configClass.getConfigurer(),
+                        new StandardSerdes(),
+                        getOkaeriConfigFileOptions(configClass.getClass()).update(),
+                        true));
+    }
+
+
+    /**
+     * Get the file name of a config class
+     * @param configClass The class of the config to get the file name of
+     * @return The file name
+     * @param <T> The type of the config
+     */
+    public static <T extends AbstractOkaeriConfigFile> String getFileName(Class<T> configClass) {
+        OkaeriConfigFileOptions options = getOkaeriConfigFileOptions(configClass);
+
+        if (!options.useLangFileName()) {
+            return options.value();
+        } else {
+            return "langs/" + TranslationManager.getInstance().getActiveTranslation().getFilename();
+        }
+    }
+
+
+
+    /**
+     * Create a new config instance with a custom file name, configurer, serdes pack, and puts it in the LOADED_CONFIGS map
+     * @param configClass The class of the config to create
+     * @param fileName The name of the file to bind the config to
+     * @param configurer The configurer to use
+     * @param serdesPack The serdes pack to use
+     * @return The new config instance
+     * @param <T> The type of the config
+     */
+    private static <T extends AbstractOkaeriConfigFile> T createConfig(Class<T> configClass, String fileName, Configurer configurer, OkaeriSerdesPack serdesPack, boolean update, boolean bind) {
+        T instance = eu.okaeri.configs.ConfigManager.create(configClass, (it) -> {
+            it.withConfigurer(configurer, serdesPack);
+            it.withRemoveOrphans(false); // Just leaving this off for now
+            if (bind) {
```

Comment:
why? (bind)

Reply:
I was trying to find a way I could create a blank **AbstractConfigFile** with just its declared defaults. In an unpushed commit (because my schools internet is awful), I was able to do this and this bind option has removed.

The reason I want an empty/completely default file is because of the NPEs and unhandled exceptions that will occur without a file being returned

Reply:
good point, but I think we're missing some benefits of using okaeri, e.g. migration

Reply:
wrong comment, lmao

Reply:
I removed this param option in the most recent commit, so I'll mark this as resolved